core/modules/loader/Central.h

```diff
@@ -0,0 +1,133 @@
+// -*- LSST-C++ -*-
+/*
+ * LSST Data Management System
+ * Copyright 2018 LSST Corporation.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ *
+ */
+#ifndef LSST_QSERV_LOADER_CENTRAL_H_
+#define LSST_QSERV_LOADER_CENTRAL_H_
+
+// system headers
+#include <boost/bind.hpp>
+#include <boost/asio.hpp>
+#include <thread>
+#include <vector>
+
+// Qserv headers
+#include "loader/ClientServer.h"
+#include "loader/MasterServer.h"
+#include "loader/MWorkerList.h"
+#include "loader/WWorkerList.h"
+#include "loader/WorkerServer.h"
+#include "proto/loader.pb.h"
+#include "util/ThreadPool.h"
+
+
+
+namespace lsst {
+namespace qserv {
+namespace loader {
+
+/// TODO add fileId and row to this so it can be checked in _workerKeyInsertReq
+struct ChunkSubchunk {
+    ChunkSubchunk(int chunk_, int subchunk_) : chunk(chunk_), subchunk(subchunk_) {}
+    int const chunk;
+    int const subchunk;
+    friend std::ostream& operator<<(std::ostream& os, ChunkSubchunk csc);
+};
+
+
+/// This class is 'central' to the execution of the program, and must be around
+/// until the bitter end. As such, it can be accessed by normal pointers.
+/// This class is central to loader workers and the master.
+/// It is the base class for CentralWorker, CentralMaster, and CentralClient
+class Central {
```

Comment:
My general comments on this class:
1) methods needs to be documented
2) inline implementations needs to be moved into the CC files to make the class's contract more observable. 
3) the class needs explicitly deleted copy c-tor and assignment operator:
```
Central(Central const&) = delete;
Central& operator=(Central const&) = delete;
```
4) If this is the base class for a family of other classes then its constructor(s) should be made `protected`.